packages/repository/src/model.ts

```diff
@@ -53,11 +48,18 @@ export class ModelDefinition {
 
   constructor(
     name: string,
-    properties?: {[name: string]: PropertyDefinition},
+    properties?: {[name: string]: PropertyDefinition | PropertyType},
```

Comment:
I wonder if we should make `ModelDefinition` an interface to be consistent with `PropertyDefinition`. If so, we probably need to have a default implementation of `ModelDefinition`.

Reply:
Well, I think there are two different entities here:

 1. A normalized model definition where each property definition has `type` property
 2. DSL for defining models, where properties can be defined using a short-hand form `{email: 'string', age: 'number'}`

I think it's ok to use current `ModelDefinition` class as the entity representing normalized model definition.

I guess it makes sense to introduce an interface to describe the DSL for defining models.

Thoughts?

Reply:
See 33a087b2be61